### PR TITLE
 Ensure PReP as primary (bsc#1082468) 

### PR DESCRIPTION
**Pull request by ancorgs**
Implements https://trello.com/c/LKmwH0cb/286-3-sles15-p1-1082468-sles15-beta7-error-during-yast-installation-bootloader-prep

- General support to enforce primary planned partitions.
- Ensures the new PReP partition suggested by BootRequirementsChecker to be primary.
- Updated documentation.

### CODE REVIEW

File: src/lib/y2storage/volume_specification_builder.rb [hunk 1]
```diff
@@ -156,7 +156,6 @@ def fallback_for_prep
         v.min_size = DiskSize.MiB(2)
         v.desired_size = DiskSize.MiB(4)
         v.max_size = DiskSize.MiB(8)
-        # So far we are always using msdos partition ids
```
Comment (jreidinger) [inline src/lib/y2storage/volume_specification_builder.rb]: why primary is not here, but at boot requirements? I have to say it is a bit confusing to have it at two places. What is rationale?

Comment (ancorgs) [reply src/lib/y2storage/volume_specification_builder.rb]: `VolumeSpecification` and `Planned::Partition` are in fact different beasts. The first one can represent both a LV and a partition. It's the OOP direct representation of the volumes as they are specified in the control file.

A `Planned::Partition` is a more down-to-earth representation of something that is indeed being to be created, and contains additional information that is only relevant for partitions.

Comment (ancorgs) [reply src/lib/y2storage/volume_specification_builder.rb]: BTW, in the past we hadn't `Planned::Partition` and `Planned::LvmLv`, we did the whole process with a more general `PlannedVolume` class... and the code turned to be VERY hard to follow. 

Creating more concrete and specialized `Planned::Partition` and `Planned::LvmLv` objects helped a lot making understandable code.
